ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -598,6 +605,25 @@ public void validate() {
             + System.lineSeparator() + KAFKA_VALUE_FORMAT_LIMITATION_DETAILS);
       }
     }
+
+    private void addSelectItem(final Expression exp, final ColumnName columnName) {
+      final Set<ColumnRef> columnRefs = new HashSet<>();
+      final TraversalExpressionVisitor<Void> visitor = new TraversalExpressionVisitor<Void>() {
```

Comment:
It would be nice if TraversalExpressionVisitor could be changed in some way such that this logic could be expressed as lambda?

Reply:
Not sure that would be possible, given there are many methods that a subclass may choose to overload.

I guess some kind of builder that handled lambdas might be possible, but not sure on the ROI ... and definitely out of scope for this PR! ;)